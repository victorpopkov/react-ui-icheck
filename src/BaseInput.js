/* eslint-disable react/no-unused-prop-types, react/no-unused-state */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import mobile from 'is-mobile';
import Input from './components/Input'; // eslint-disable-line sort-imports
import Label from './components/Label';
import deprecated from './helpers/deprecated';

class BaseInput extends Component {
  static propTypes = {
    activeClass: deprecated(PropTypes.string, 'activeClassName'),
    activeClassName: PropTypes.string,
    aria: PropTypes.bool,
    checked: PropTypes.bool,
    checkedClass: deprecated(PropTypes.string, 'checkedClassName'),
    checkedClassName: PropTypes.string,
    children: PropTypes.node,
    className: PropTypes.string,
    defaultChecked: PropTypes.bool,
    disabled: PropTypes.bool,
    disabledClass: deprecated(PropTypes.string, 'disabledClassName'),
    disabledClassName: PropTypes.string,
    focusClass: deprecated(PropTypes.string, 'focusClassName'),
    focusClassName: PropTypes.string,
    hoverClass: deprecated(PropTypes.string, 'hoverClassName'),
    hoverClassName: PropTypes.string,
    id: PropTypes.string,
    increaseArea: PropTypes.string,
    indeterminate: PropTypes.bool,
    indeterminateClass: deprecated(PropTypes.string, 'indeterminateClassName'),
    indeterminateClassName: PropTypes.string,
    inheritClass: deprecated(PropTypes.bool, 'inheritClassName'),
    inheritClassName: PropTypes.bool,
    inheritID: PropTypes.bool,
    insert: PropTypes.node,
    label: PropTypes.node,
    labelClassName: PropTypes.string,
    labelHover: PropTypes.bool,
    labelHoverClass: PropTypes.string,
    labelTag: PropTypes.string,
    labelTagClassName: PropTypes.string,
    name: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onMouseDown: PropTypes.func,
    onMouseLeave: PropTypes.func,
    onMouseUp: PropTypes.func,
    onTouchEnd: PropTypes.func,
    onTouchStart: PropTypes.func,
  };

  static defaultProps = {
    activeClass: null,
    activeClassName: 'active',
    aria: false,
    checked: false,
    checkedClass: null,
    checkedClassName: 'checked',
    children: null,
    className: null,
    defaultChecked: false,
    disabled: false,
    disabledClass: null,
    disabledClassName: 'disabled',
    focusClass: null,
    focusClassName: 'focus',
    hoverClass: null,
    hoverClassName: 'hover',
    id: null,
    increaseArea: '',
    indeterminate: true,
    indeterminateClass: null,
    indeterminateClassName: 'indeterminate',
    inheritClass: false,
    inheritClassName: false,
    inheritID: false,
    insert: null,
    label: null,
    labelClassName: null,
    labelHover: true,
    labelHoverClass: 'hover',
    labelTag: 'span',
    labelTagClassName: null,
    name: null,
    type: null,
    value: '',
    onBlur: null,
    onChange: null,
    onFocus: null,
    onMouseDown: null,
    onMouseLeave: null,
    onMouseUp: null,
    onTouchEnd: null,
    onTouchStart: null,
  };

  constructor(props) {
    super(props);

    this.handleHover = this.handleHover.bind(this);

    const { checked } = props;

    this.state = {
      checked: checked || false,
      hovered: false,
      isMobile: mobile(),
    };
  }

  componentWillReceiveProps(nextProps) {
    const { checked } = this.props;
    const { checked: nextChecked } = nextProps;

    if ('checked' in nextProps && checked !== nextChecked) {
      this.setState({ checked: nextChecked });
    }
  }

  handleHover(event, isHovered) {
    this.setState({ hovered: isHovered });
  }

  renderInput(props) {
    const {
      activeClass,
      activeClassName,
      aria,
      checkedClass,
      checkedClassName,
      className,
      defaultChecked,
      disabled,
      disabledClass,
      disabledClassName,
      focusClass,
      focusClassName,
      hoverClass,
      hoverClassName,
      id,
      increaseArea,
      indeterminate,
      indeterminateClass,
      indeterminateClassName,
      inheritClass,
      inheritClassName,
      inheritID,
      insert,
      name,
      onBlur,
      onChange,
      onFocus,
      value,

      checkboxClass, // eslint-disable-line react/prop-types
      checked,
      children,
      label,
      labelClassName,
      labelHover,
      labelHoverClass,
      labelTag,
      labelTagClassName,
      onMouseDown,
      onMouseLeave,
      onMouseUp,
      onTouchEnd,
      onTouchStart,
      radioClass, // eslint-disable-line react/prop-types
      type,

      ...other
    } = this.props;
    const { checked: stateChecked, hovered, isMobile } = this.state;

    return (
      <Input
        {...other}
        {...props}
        activeClassName={activeClass || activeClassName}
        aria={aria}
        checked={stateChecked}
        checkedClassName={checkedClass || checkedClassName}
        className={className}
        defaultChecked={defaultChecked}
        disabled={disabled}
        disabledClassName={disabledClass || disabledClassName}
        focusClassName={focusClass || focusClassName}
        hoverClassName={hoverClass || hoverClassName}
        hovered={hovered}
        id={id}
        increaseArea={increaseArea}
        indeterminate={indeterminate}
        indeterminateClassName={indeterminateClass || indeterminateClassName}
        inheritClassName={inheritClass || inheritClassName}
        inheritID={inheritID}
        insert={insert}
        isMobile={isMobile}
        name={name}
        ref={(c) => { this.input = c; }}
        value={value}
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
      />
    );
  }

  render() {
    const {
      children,
      disabled,
      label,
      labelClassName,
      labelHover,
      labelTag,
      labelTagClassName,
    } = this.props;
    const { isMobile } = this.state;

    if (!label && !children) {
      return this.renderInput();
    }

    return (
      <Label
        disabled={disabled}
        handleHover={this.handleHover}
        isMobile={isMobile}
        label={label}
        labelChildren={children}
        labelClassName={labelClassName}
        labelHover={labelHover}
        labelTag={labelTag}
        labelTagClassName={labelTagClassName}
      >
        {this.renderInput()}
      </Label>
    );
  }
}

export default BaseInput;
