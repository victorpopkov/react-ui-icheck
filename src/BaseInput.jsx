import React, { Component } from 'react';
import PropTypes from 'prop-types';
import mobile from 'is-mobile';
import Input from './components/Input'; // eslint-disable-line sort-imports
import Label from './components/Label';

class BaseInput extends Component {
  constructor(props) {
    super(props);

    this.handleHover = this.handleHover.bind(this);

    const { checked } = props;

    this.state = {
      hovered: false,
      isMobile: mobile(),
      prevPropChecked: props.checked,
      checked,
    };
  }

  static getDerivedStateFromProps(props, state) {
    if (state.prevPropChecked !== props.checked) {
      return {
        checked: props.checked,
        prevPropChecked: props.checked,
      };
    }

    return null;
  }

  handleHover(event, hovered) {
    this.setState({ hovered });
  }

  renderInput(props) {
    const {
      activeClassName,
      aria,
      checkedClassName,
      className,
      disabled,
      disabledClassName,
      focusClassName,
      hoverClassName,
      id,
      increaseArea,
      indeterminate,
      indeterminateClassName,
      inheritClassName,
      inheritID,
      insert,
      name,
      onBlur,
      onChange,
      onFocus,
      value,

      checkboxClass, // eslint-disable-line react/prop-types
      checked,
      children,
      label,
      labelClassName,
      labelHover,
      labelHoverClass,
      labelTag,
      labelTagClassName,
      onMouseDown,
      onMouseLeave,
      onMouseUp,
      onTouchEnd,
      onTouchStart,
      radioClass, // eslint-disable-line react/prop-types
      type,

      ...other
    } = this.props;
    const { checked: stateChecked, hovered, isMobile } = this.state;

    return (
      <Input
        {...other}
        {...props}
        activeClassName={activeClassName}
        aria={aria}
        checked={stateChecked}
        checkedClassName={checkedClassName}
        className={className}
        disabled={disabled}
        disabledClassName={disabledClassName}
        focusClassName={focusClassName}
        hoverClassName={hoverClassName}
        hovered={hovered}
        id={id}
        increaseArea={increaseArea}
        indeterminate={indeterminate}
        indeterminateClassName={indeterminateClassName}
        inheritClassName={inheritClassName}
        inheritID={inheritID}
        insert={insert}
        isMobile={isMobile}
        name={name}
        ref={(c) => {
          // eslint-disable-next-line react/no-unused-class-component-methods
          this.input = c;
        }}
        value={value}
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
      />
    );
  }

  render() {
    const {
      children,
      disabled,
      label,
      labelClassName,
      labelHover,
      labelTag,
      labelTagClassName,
    } = this.props;
    const { isMobile } = this.state;

    if (!label && !children) {
      return this.renderInput();
    }

    return (
      <Label
        disabled={disabled}
        handleHover={this.handleHover}
        isMobile={isMobile}
        label={label}
        labelChildren={children}
        labelClassName={labelClassName}
        labelHover={labelHover}
        labelTag={labelTag}
        labelTagClassName={labelTagClassName}
      >
        {this.renderInput()}
      </Label>
    );
  }
}

BaseInput.propTypes = {
  activeClassName: PropTypes.string,
  aria: PropTypes.bool,
  checked: PropTypes.bool,
  checkedClassName: PropTypes.string,
  children: PropTypes.node,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  disabledClassName: PropTypes.string,
  focusClassName: PropTypes.string,
  hoverClassName: PropTypes.string,
  id: PropTypes.string,
  increaseArea: PropTypes.string,
  indeterminate: PropTypes.bool,
  indeterminateClassName: PropTypes.string,
  inheritClassName: PropTypes.bool,
  inheritID: PropTypes.bool,
  insert: PropTypes.node,
  label: PropTypes.node,
  labelClassName: PropTypes.string,
  labelHover: PropTypes.bool,
  labelHoverClass: PropTypes.string,
  labelTag: PropTypes.string,
  labelTagClassName: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  onBlur: PropTypes.func,
  onChange: PropTypes.func,
  onFocus: PropTypes.func,
  onMouseDown: PropTypes.func,
  onMouseLeave: PropTypes.func,
  onMouseUp: PropTypes.func,
  onTouchEnd: PropTypes.func,
  onTouchStart: PropTypes.func,
};

BaseInput.defaultProps = {
  activeClassName: 'active',
  aria: false,
  checked: false,
  checkedClassName: 'checked',
  children: null,
  className: null,
  disabled: false,
  disabledClassName: 'disabled',
  focusClassName: 'focus',
  hoverClassName: 'hover',
  id: null,
  increaseArea: '',
  indeterminate: true,
  indeterminateClassName: 'indeterminate',
  inheritClassName: false,
  inheritID: false,
  insert: null,
  label: null,
  labelClassName: null,
  labelHover: true,
  labelHoverClass: 'hover',
  labelTag: 'span',
  labelTagClassName: null,
  name: null,
  type: null,
  value: '',
  onBlur: null,
  onChange: null,
  onFocus: null,
  onMouseDown: null,
  onMouseLeave: null,
  onMouseUp: null,
  onTouchEnd: null,
  onTouchStart: null,
};

export default BaseInput;
