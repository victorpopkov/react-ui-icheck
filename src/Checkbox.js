import React, { Component } from 'react';
import PropTypes from 'prop-types';
import mobile from 'is-mobile';
import Input from './components/Input'; // eslint-disable-line sort-imports
import Label from './components/Label';
import deprecated from './helpers/deprecated';

class Checkbox extends Component {
  static propTypes = {
    activeClass: deprecated(PropTypes.string, 'activeClassName'),
    activeClassName: PropTypes.string,
    aria: PropTypes.bool,
    checkboxClass: deprecated(PropTypes.string, 'checkboxClassName'),
    checkboxClassName: PropTypes.string,
    checked: PropTypes.bool,
    checkedClass: deprecated(PropTypes.string, 'checkedClassName'),
    checkedClassName: PropTypes.string,
    children: PropTypes.node,
    className: PropTypes.string,
    defaultChecked: PropTypes.bool,
    determinateClass: PropTypes.string,
    disabled: PropTypes.bool,
    disabledClass: deprecated(PropTypes.string, 'disabledClassName'),
    disabledClassName: PropTypes.string,
    focusClass: deprecated(PropTypes.string, 'focusClassName'),
    focusClassName: PropTypes.string,
    hoverClass: deprecated(PropTypes.string, 'hoverClassName'),
    hoverClassName: PropTypes.string,
    id: PropTypes.string,
    increaseArea: PropTypes.string,
    indeterminate: PropTypes.bool,
    indeterminateClass: PropTypes.string,
    inheritClass: PropTypes.bool,
    inheritID: PropTypes.bool,
    insert: PropTypes.node,
    label: PropTypes.node,
    labelClassName: PropTypes.string,
    labelHover: PropTypes.bool,
    labelHoverClass: PropTypes.string,
    labelTag: PropTypes.string,
    labelTagClassName: PropTypes.string,
    name: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onMouseDown: PropTypes.func,
    onMouseLeave: PropTypes.func,
    onMouseUp: PropTypes.func,
    onTouchEnd: PropTypes.func,
    onTouchStart: PropTypes.func,
  };

  static defaultProps = {
    activeClass: 'active',
    activeClassName: 'active',
    aria: false,
    checkboxClass: 'icheckbox',
    checkboxClassName: 'icheckbox',
    checked: false,
    checkedClass: 'checked',
    checkedClassName: 'checked',
    children: null,
    className: null,
    defaultChecked: false,
    determinateClass: null,
    disabled: false,
    disabledClass: 'disabled',
    disabledClassName: 'disabled',
    focusClass: 'focus',
    focusClassName: 'focus',
    hoverClass: 'hover',
    hoverClassName: 'hover',
    id: null,
    increaseArea: '',
    indeterminate: true,
    indeterminateClass: 'indeterminate',
    inheritClass: false,
    inheritID: false,
    insert: null,
    label: null,
    labelClassName: null,
    labelHover: true,
    labelHoverClass: 'hover',
    labelTag: 'span',
    labelTagClassName: null,
    name: null,
    type: null,
    value: '',
    onBlur: null,
    onChange: null,
    onFocus: null,
    onMouseDown: null,
    onMouseLeave: null,
    onMouseUp: null,
    onTouchEnd: null,
    onTouchStart: null,
  };

  constructor(props) {
    super(props);

    this.handleHover = this.handleHover.bind(this);

    const { checked } = props;

    this.state = {
      checked: checked || false,
      hovered: false,
      isMobile: mobile(),
    };
  }

  componentWillReceiveProps(nextProps) {
    const { checked } = this.props;
    const { checked: nextChecked } = nextProps;

    if ('checked' in nextProps && checked !== nextChecked) {
      this.setState({
        checked: nextChecked,
      });
    }
  }

  handleHover(event, isHovered) {
    this.setState({
      hovered: isHovered,
    });
  }

  renderInput() {
    const {
      activeClass,
      activeClassName,
      aria,
      checkboxClass,
      checkboxClassName,
      checkedClass,
      checkedClassName,
      className,
      defaultChecked,
      disabled,
      disabledClass,
      disabledClassName,
      focusClass,
      focusClassName,
      hoverClass,
      hoverClassName,
      id,
      increaseArea,
      indeterminate,
      indeterminateClass,
      inheritClass,
      inheritID,
      insert,
      name,
      onBlur,
      onChange,
      onFocus,
      value,

      checked,
      children,
      determinateClass,
      label,
      labelClassName,
      labelHover,
      labelHoverClass,
      labelTag,
      labelTagClassName,
      onMouseDown,
      onMouseLeave,
      onMouseUp,
      onTouchEnd,
      onTouchStart,
      type,

      ...other
    } = this.props;
    const { checked: stateChecked, hovered, isMobile } = this.state;

    return (
      <Input
        {...other}
        activeClass={activeClass}
        activeClassName={activeClassName}
        aria={aria}
        checkboxClass={checkboxClass}
        checkboxClassName={checkboxClassName}
        checked={stateChecked}
        checkedClass={checkedClass}
        checkedClassName={checkedClassName}
        className={className}
        defaultChecked={defaultChecked}
        disabled={disabled}
        disabledClass={disabledClass}
        disabledClassName={disabledClassName}
        focusClass={focusClass}
        focusClassName={focusClassName}
        hoverClass={hoverClass}
        hoverClassName={hoverClassName}
        hovered={hovered}
        id={id}
        increaseArea={increaseArea}
        indeterminate={indeterminate}
        indeterminateClass={indeterminateClass}
        inheritClass={inheritClass}
        inheritID={inheritID}
        inputType="checkbox"
        insert={insert}
        isMobile={isMobile}
        name={name}
        ref={(c) => { this.input = c; }}
        value={value}
        onBlur={onBlur}
        onChange={onChange}
        onFocus={onFocus}
      />
    );
  }

  render() {
    const {
      children,
      disabled,
      label,
      labelClassName,
      labelHover,
      labelTag,
      labelTagClassName,
    } = this.props;
    const { isMobile } = this.state;

    if (!label && !children) {
      return this.renderInput();
    }

    return (
      <Label
        disabled={disabled}
        handleHover={this.handleHover}
        isMobile={isMobile}
        label={label}
        labelChildren={children}
        labelClassName={labelClassName}
        labelHover={labelHover}
        labelTag={labelTag}
        labelTagClassName={labelTagClassName}
      >
        {this.renderInput()}
      </Label>
    );
  }
}

export default Checkbox;
