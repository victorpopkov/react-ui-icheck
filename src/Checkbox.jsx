import PropTypes from 'prop-types';
import BaseInput from './BaseInput'; // eslint-disable-line sort-imports

class Checkbox extends BaseInput {
  renderInput() {
    const { checkboxClass, checkboxClassName } = this.props;

    return super.renderInput({
      checkboxClassName: checkboxClass || checkboxClassName,
      inputType: 'checkbox',
    });
  }
}

Checkbox.propTypes = {
  ...BaseInput.propTypes,
  checkboxClassName: PropTypes.string,
};

Checkbox.defaultProps = {
  ...BaseInput.defaultProps,
  checkboxClass: null,
  checkboxClassName: 'icheckbox',
};

export default Checkbox;
