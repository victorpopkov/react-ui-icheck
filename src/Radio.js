import PropTypes from 'prop-types';
import BaseInput from './BaseInput'; // eslint-disable-line sort-imports
import deprecated from './helpers/deprecated';

class Radio extends BaseInput {
  static propTypes = {
    ...BaseInput.propTypes,
    radioClass: deprecated(PropTypes.string, 'radioClassName'),
    radioClassName: PropTypes.string,
  };

  static defaultProps = {
    ...BaseInput.defaultProps,
    radioClass: null,
    radioClassName: 'iradio',
  };

  renderInput() {
    const { radioClass, radioClassName } = this.props;

    return super.renderInput({
      inputType: 'radio',
      radioClassName: radioClass || radioClassName,
    });
  }
}

export default Radio;
