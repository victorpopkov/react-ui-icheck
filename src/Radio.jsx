import PropTypes from 'prop-types';
import BaseInput from './BaseInput'; // eslint-disable-line sort-imports

class Radio extends BaseInput {
  renderInput() {
    const { radioClass, radioClassName } = this.props;

    return super.renderInput({
      inputType: 'radio',
      radioClassName: radioClass || radioClassName,
    });
  }
}

Radio.propTypes = {
  ...BaseInput.propTypes,
  radioClassName: PropTypes.string,
};

Radio.defaultProps = {
  ...BaseInput.defaultProps,
  radioClass: null,
  radioClassName: 'iradio',
};

export default Radio;
