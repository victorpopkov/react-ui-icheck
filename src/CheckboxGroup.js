import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Checkbox from './Checkbox'; // eslint-disable-line sort-imports

class CheckboxGroup extends Component {
  renderCheckbox(props) {
    const { checkboxWrapClassName, checkboxWrapTag } = this.props;
    const { value, ...other } = props;

    const checkbox = (
      <Checkbox
        {...other}
        key={value}
        value={value}
      />
    );

    if (checkboxWrapTag) {
      const WrapTag = checkboxWrapTag;

      return (
        <WrapTag className={checkboxWrapClassName}>
          {checkbox}
        </WrapTag>
      );
    }

    return checkbox;
  }

  render() {
    const {
      children,
      className,

      checkboxWrapClassName,
      checkboxWrapTag,

      ...other
    } = this.props;

    const options = React.Children.map(
      children,
      (option) => this.renderCheckbox(option.props),
      this
    );

    return (
      <div {...other} className={className}>
        {options}
      </div>
    );
  }
}

CheckboxGroup.propTypes = {
  children: PropTypes.node.isRequired,
  checkboxWrapClassName: PropTypes.string,
  checkboxWrapTag: PropTypes.string,
  className: PropTypes.string,
};

CheckboxGroup.defaultProps = {
  checkboxWrapClassName: null,
  checkboxWrapTag: null,
  className: null,
};

export default CheckboxGroup;
