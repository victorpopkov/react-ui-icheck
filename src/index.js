import Checkbox from './Checkbox';
import CheckboxGroup from './CheckboxGroup';
import Radio from './Radio';
import RadioGroup from './RadioGroup';

export { Checkbox, CheckboxGroup, Radio, RadioGroup };
