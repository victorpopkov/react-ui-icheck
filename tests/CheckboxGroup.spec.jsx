import React from 'react';
import expect from 'expect';
import { mount } from 'enzyme';
import { Checkbox, CheckboxGroup } from '../src';

describe('CheckboxGroup component', () => {
  let wrapper;

  beforeEach(() => {
    console.warn = jest.fn();
  });

  describe('when no props are passed', () => {
    describe('as mount', () => {
      beforeEach(() => {
        wrapper = mount(
          <CheckboxGroup>
            <Checkbox />
          </CheckboxGroup>,
        );
      });

      it('should not call the deprecated console.warn()', () => {
        expect(console.warn).toHaveBeenCalledTimes(0);
      });

      it('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
      });

      it('should match the default props snapshot', () => {
        expect(wrapper.props()).toMatchSnapshot();
      });
    });
  });

  describe('and the `checkboxWrapClassName` prop is `test`', () => {
    describe('and the `checkboxWrapTag` prop is `div`', () => {
      describe('as mount', () => {
        beforeEach(() => {
          wrapper = mount(
            <CheckboxGroup checkboxWrapClassName="test" checkboxWrapTag="div">
              <Checkbox />
            </CheckboxGroup>,
          );
        });

        it('should match the snapshot', () => {
          expect(wrapper).toMatchSnapshot();
        });
      });
    });
  });
});
