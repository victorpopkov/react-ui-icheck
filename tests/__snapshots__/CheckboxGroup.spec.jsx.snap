// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckboxGroup /> when children are passed and \`checkboxWrapTag\` prop is "span" and \`checkboxWrapClassName\` prop is "test" should match the snapshot 1`] = `
<div>
  <div>
    <span
      class="test"
    >
      <div
        class="icheckbox indeterminate"
      >
        <input
          style="opacity: 0; position: absolute;"
          type="checkbox"
          value=""
        />
        <ins
          class="iCheck-helper"
          role="presentation"
          style="position: absolute; top: 0%; left: 0%; display: block; width: 100%; height: 100%; margin: 0px; padding: 0px; background: rgb(255, 255, 255); border: 0px; opacity: 0; cursor: pointer;"
        />
      </div>
    </span>
  </div>
</div>
`;

exports[`<CheckboxGroup /> when children are passed and \`className\` prop is "test" should match the snapshot 1`] = `
<div>
  <div
    class="test"
  >
    <div
      class="icheckbox indeterminate"
    >
      <input
        style="opacity: 0; position: absolute;"
        type="checkbox"
        value=""
      />
      <ins
        class="iCheck-helper"
        role="presentation"
        style="position: absolute; top: 0%; left: 0%; display: block; width: 100%; height: 100%; margin: 0px; padding: 0px; background: rgb(255, 255, 255); border: 0px; opacity: 0; cursor: pointer;"
      />
    </div>
  </div>
</div>
`;

exports[`<CheckboxGroup /> when children are passed should match the snapshot 1`] = `
<div>
  <div>
    <div
      class="icheckbox indeterminate"
    >
      <input
        style="opacity: 0; position: absolute;"
        type="checkbox"
        value=""
      />
      <ins
        class="iCheck-helper"
        role="presentation"
        style="position: absolute; top: 0%; left: 0%; display: block; width: 100%; height: 100%; margin: 0px; padding: 0px; background: rgb(255, 255, 255); border: 0px; opacity: 0; cursor: pointer;"
      />
    </div>
  </div>
</div>
`;
